src/main/java/org/folio/consortia/service/impl/PrimaryAffiliationAsyncServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+package org.folio.consortia.service.impl;
+
+import java.util.UUID;
+import java.util.stream.IntStream;
+
+import org.folio.consortia.config.kafka.KafkaService;
+import org.folio.consortia.domain.dto.PrimaryAffiliationEvent;
+import org.folio.consortia.domain.dto.SyncPrimaryAffiliationBody;
+import org.folio.consortia.domain.dto.SyncUser;
+import org.folio.consortia.domain.entity.TenantEntity;
+import org.folio.consortia.repository.UserTenantRepository;
+import org.folio.consortia.service.TenantService;
+import org.folio.consortia.service.PrimaryAffiliationAsyncService;
+import org.folio.consortia.service.UserTenantService;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@AllArgsConstructor
+public class PrimaryAffiliationAsyncServiceImpl implements PrimaryAffiliationAsyncService {
+
+  private final UserTenantService userTenantService;
+  private final TenantService tenantService;
+  private final KafkaService kafkaService;
+  private final UserTenantRepository userTenantRepository;
+  private final ConversionService converter;
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Override
+  @Async
+  public void createPrimaryUserAffiliations(UUID consortiumId, SyncPrimaryAffiliationBody syncPrimaryAffiliationBody) {
+    log.info("Start creating user primary affiliation for tenant {}", syncPrimaryAffiliationBody.getTenantId());
+    var tenantId = syncPrimaryAffiliationBody.getTenantId();
+    var userList = syncPrimaryAffiliationBody.getUsers();
+    try {
+      var tenantEntity = tenantService.getByTenantId(tenantId);
+      IntStream.range(0, userList.size())
+        .sequential()
+        .forEach(idx -> {
+          var user = userList.get(idx);
+          log.info("Processing users: {} of {}", idx + 1, userList.size());
+          var consortiaUserTenant = userTenantRepository.findByUserIdAndTenantId(UUID.fromString(user.getId()), tenantId)
+            .orElse(null);
+          if (consortiaUserTenant != null && consortiaUserTenant.getIsPrimary()) {
```

Comment:
sample and shadow users have the same uuid, we should primary affiliate them, we should create primary affiliation only when user_tenants table doesn't contain any records for user uuid